airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/task/DestinationTaskLauncher.kt

```diff
@@ -247,6 +245,12 @@ class DefaultDestinationTaskLauncher(
         }
     }
 
+    override suspend fun handleSetupComplete() {
```

Comment:
IMHO, not the clearest pattern here. There are some dependencies across tasks, however, it looks like they are handled in a very implicit manner.
I am also not convinced that we should start anything before the setup completes.